lake_hylia.logic

```diff
@@ -1,3 +1,77 @@
 area LakeHylia:
-  room BeanstalkArea
-  room Main
+  room BeanstalkArea:
+    node EastArea:
+      door LeftLadder
+      door RightLadder
+    node WestArea:
+      door Ladder
+    node BeanstalkFusion:
+      door BeanstalkFusion
+    WestArea <-> BeanstalkFusion: flag HyliaBeanstalkFusion
```

Comment:
We'll probably need to split this `door` (i.e. put the `exit` with the WestArea node) - but to make sure, what happens if you climb down the beanstalk without the fusion being done?

Reply:
It actually softlocks you, so entering from unfused beanstalk is impossible

Reply:
It's just going to need to be its own node in that case (so you can't go anywhere from the `exit`)